moped-editor/src/views/projects/projectView/ProjectFiles.js

```diff
@@ -212,43 +205,17 @@ const useColumns = ({
         sortable: false,
         editable: false,
         type: "actions",
-        getActions: ({ id }) => {
-          const isInEditMode = rowModesModel[id]?.mode === GridRowModes.Edit;
-          if (isInEditMode) {
-            return [
-              <GridActionsCellItem
-                icon={<CheckIcon sx={defaultEditColumnIconStyle} />}
-                label="Save"
-                sx={{
-                  color: "primary.main",
-                }}
-                onClick={handleSaveClick(id)}
-              />,
-              <GridActionsCellItem
-                icon={<CloseIcon sx={defaultEditColumnIconStyle} />}
-                label="Cancel"
-                className="textPrimary"
-                onClick={handleCancelClick(id)}
-                color="inherit"
-              />,
-            ];
-          }
-          return [
-            <GridActionsCellItem
-              icon={<EditOutlinedIcon sx={defaultEditColumnIconStyle} />}
-              label="Edit"
-              className="textPrimary"
-              onClick={handleEditClick(id)}
-              color="inherit"
-            />,
-            <GridActionsCellItem
-              icon={<DeleteOutlineIcon sx={defaultEditColumnIconStyle} />}
-              label="Delete"
-              onClick={() => handleDeleteOpen(id)}
-              color="inherit"
-            />,
-          ];
-        },
+        renderCell: ({ id }) => (
+          <DataGridActions
+            id={id}
+            requiredFields={["file_name", "file_url", "file_type"]}
```

Comment:
do yall like this or do you want to define requiredFields elsewhere in a var somewhere?

Reply:
this feels good to me. If I were to start a new table that needed an inline form, I could imagine this prop making me stop to consider what fields need are required as soon as i added this component.

I briefly thought about a way to tie to the column config but that feels like over-optimization at this point imo. None of these inline forms have a long list of required fields since we've tried to minimize side-scrolling.

Reply:
obligatory comment about non-stable object references: i think this is slightest of anti-patterns versus declaring  a `const` outside of the component 😇

Reply:
ahh yeah that is a great point, John! could you update @roseeichelmann?